src/App.jsx

```diff
@@ -1,7 +1,101 @@
+import { useState } from "react";
 import "./App.css";
+import BottomModalLayout from "./components/BottomModalLayout/BottomModalLayout";
+import Header from "./components/common/Header/Header";
+import RestaurantListLayout from "./components/RestaurantListLayout/RestaurantListLayout";
+import RestaurantDetailModal from "./components/BottomModalLayout/RestaurantDetailModal/RestaurantDetailModal";
+import AddRestaurantModal from "./components/BottomModalLayout/AddRestaurantModal/AddRestaurantModal";
+import { useEffect } from "react";
 
 function App() {
-  return <h1>Self-Paced React</h1>;
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalName, setModalName] = useState("");
+  const [selectedRestaurant, setSelectedRestaurant] = useState(null);
+  const [restaurants, setRestaurants] = useState([]);
+
+  useEffect(() => {
+    const fetchRestaurants = async () => {
+      const response = await fetch("http://localhost:3000/restaurants");
+      const data = await response.json();
+      setRestaurants(data);
+    };
+
+    fetchRestaurants();
+  }, []);
+
+  const openRestaurantDetailModal = ({ restaurant }) => {
```

Comment:
```suggestion
  const openRestaurantDetailModal = (restaurant) => {
```
인자로 들어오는 `restaurant`를 객체로 받는 이유가 있으신가요??🤔

Reply:
일단 확장성이랑 명확성을 고려해서 이렇게 작성해두긴 했어요!
그냥 단순 매개변수 인자로 받아도 충분한 상황이지만, 나중에 전달해야하는 데이터가 늘어날 수도 있고,
리액트의 props 패턴은 하나의 객체만 받으니까 그 느낌과의 통일성을 노린 부분도 있어요!

그리고 이 함수를 호출하는 곳에서도 openRestaurantDetailModal(restaurant) 라고 쓰는 것보다 openRestaurantDetailModal({restaurant}) 라고쓰면  restaurant 라는 키가 필요한 함수구나라고 읽혀져서,
restaurant가 음식점 이름 string일지, 객체의 데이터값일지 모호한 지점이 조금 더 명확해지는 느낌...?이라고 생각했어요!

Reply:
> 리액트의 props 패턴은 하나의 객체만 받으니까 그 느낌과의 통일성을 노린 부분도 있어요!

오! 저는 오히려 함수형 컴포넌트(UI)랑 일반 함수(로직)랑 목적이 다르다고 생각해서 통일하기 보다는 차이점을 두는 것 같아요🙃

그래서 컨벤션도 컴포넌트 선언은 `function 키워드` 사용하고 일반 함수는 `화살표 함수`를 사용하는 것 같아요!
```tsx
// 이런 느낌으로?
function Component({ onClick, data }){
  const sum = (a, b) => a + b;
}
```

> 그리고 이 함수를 호출하는 곳에서도 openRestaurantDetailModal(restaurant) 라고 쓰는 것보다 openRestaurantDetailModal({restaurant}) 라고쓰면 restaurant 라는 키가 필요한 함수구나라고 읽혀져서,
restaurant가 음식점 이름 string일지, 객체의 데이터값일지 모호한 지점이 조금 더 명확해지는 느낌...?이라고 생각했어요!

단일 인자일 때는 객체보다는 그냥 쓰는게 좋은 것 같은데, 인자가 여러 개인 경우에는 카멜 말처럼 객체로 받고 객체의 키로 어떤 값이 와야한다는걸 명시하는게 좋은 것 같네요👍👍👍

Reply:
!!!!😳
> 오! 저는 오히려 함수형 컴포넌트(UI)랑 일반 함수(로직)랑 목적이 다르다고 생각해서 통일하기 보다는 차이점을 두는 것 같아요🙃

완전 일리있는데요...? 호오오오오오......목적이 다르니까 오히려 차이점을 둔다...인상깊었어요!!! 써밋은 딱 이유 있는 기준을 가지고 컨벤션을 형성하는 것 같아서 배울점이 넘 많은 거 같아요!!! 레벨2도 써피티 자주 써야겠ㄷ ㅋㅋㅋㅋㅎ

Reply:
ㅋㅋㅋㅋ근데 다른분들은 함수형 컴포넌트도 함수고, 일반 함수도 함수여서 `function 키워드` or `화살표 함수`로 통일하는 경우도 많아서 취향차이인 것 같긴해여🙃